sample/samplers.go

```diff
@@ -165,22 +165,21 @@ func NewSampler(temperature float32, topK int, topP float32, minP float32, seed
 }
 
 type Grammar struct {
-	vocab   *Vocab
-	grammar string
-	sampler *llama.Sampler
+	vocab   *model.Vocabulary
+	grammar *llama.Grammar
 }
 
-func NewGrammar(vocab *Vocab, grammar string) (*Grammar, error) {
-	v, err := vocab.Load()
-	if err != nil {
-		return nil, err
+func NewGrammar(vocab *model.Vocabulary, grammarStr string) (*Grammar, error) {
```

Comment:
Is a unit test possible here now that we can pass an Ollama grammar (those are easier to create for testing)

Reply:
now that it is an ollama vocab this is kinda hard to do